### PR TITLE
Fix the default for async finalizer

### PR DESCRIPTION
**Pull request by pditommaso**
The PR fixes the setting for the `NXF_ENABLE_ASYNC_FINALIZER` env variable. 

The expression `'some string' as boolean` is always true unless the string is null or empty. 

Therefore the setting is always enabled unless `NXF_ENABLE_ASYNC_FINALIZER` is set to an empty string.

### CODE REVIEW

**Comment by pditommaso** [comment]
Wonder if it should be set to `true` by default at this point

**Comment by bentsherman** [comment]
Might be better to make it true by default and allow people to disable it if it breaks something for them. We'll get much better testing feedback that way

File: modules/nextflow/src/main/groovy/nextflow/processor/TaskPollingMonitor.groovy [hunk 1]
```diff
@@ -118,7 +118,7 @@ class TaskPollingMonitor implements TaskMonitor {
     @Lazy
     private ExecutorService finalizerPool = { session.taskFinalizerExecutorService() }()
 
-    private boolean enableAsyncFinalizer = SysEnv.get('NXF_ENABLE_ASYNC_FINALIZER','false') as boolean
+    private boolean enableAsyncFinalizer = SysEnv.getBool('NXF_ENABLE_ASYNC_FINALIZER',false)
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/processor/TaskPollingMonitor.groovy:121]:
```suggestion
    private boolean enableAsyncFinalizer = SysEnv.getBool('NXF_ENABLE_ASYNC_FINALIZER',true)
```

**Comment by pditommaso** [comment]
ok

**Pull request merged**